audmodel/core/lock.py

```diff
@@ -0,0 +1,78 @@
+from contextlib import ExitStack
+from contextlib import contextmanager
+import os
+import warnings
+
+from filelock import FileLock
+from filelock import Timeout
+
+import audeer
+
+
+@contextmanager
+def lock(
+    paths: list[str],
+    *,
+    timeout: float = 86400,
+    warning_timeout: float = 2,
```

Comment:
Giving a warning after 2 seconds seems pretty fast, compared to that the default value of 24 h appears quite long. But in `audb` we also use 24h as default, so I guess it's ok.

Reply:
Now I got it. I changed now to `waiting_timeout=0` instead of removing it, so we can still use it in the tests or when we decide to move the function to another package. This means we do now immediately show a warning message if we cannot acquire a lock.

In addition, I set the default for raising an error to `timeout=10`, and only change it to a larger value (24 h) inside `audmodel.load()`.

Reply:
Ah or is the use-case that you set the warning timeout > timeout to completely suppress the warning?